src/twisted/application/test/test_service.py

```diff
@@ -23,13 +23,36 @@ class AlmostService(object):
     """
     Implement IService except for the attributes.
     """
+
+    def __init__(self, name, parent, running):
+        self.name = name
+        self.parent = parent
+        self.running = running
+
+    def makeInvalidByDeletingName(self):
+        """
+        Probably not a wise method to call.
+        """
+        del self.name
+
+    def makeInvalidByDeletingParent(self):
+        """
+        Probably not a wise method to call.
+        """
+        del self.parent
+
+    def makeInvalidByDeletingRunning(self):
+        """
+        Probably not a wise method to call.
```

Comment:
Why does it exist, then? Please write a doc string that explains the purpose of the method.